client.go

```diff
@@ -84,6 +91,13 @@ func (cl *Client) Do(c context.Context, req *Request) (*Response, error) {
 			return nil, err
 		}
 
+		// if we used a multipart form, we need to check for an error from the goroutine
+		if req.optMultiPartForm && i == 1 {
```

Comment:
Not sure on this, but wouldn't this logic get hit less if you put the `i == 1` as the first statement?